regions_www/m/server.R

```diff
@@ -19,804 +19,1071 @@
         query[['r']]
-      } else if (exists("starting_region")) {
-        starting_region
       } else {
-        "west-yorkshire"
+        "isle-of-wight"
       }
-      region$data_dir <- file.path(data_dir_root, region$current)
-      region$all_trips <- dir.exists(file.path(data_dir_root, region$current , 'all-trips'))
     }
 
+    # Notify browser to update URL to reflect new region
     session$sendCustomMessage("regionchange", region$current)
-    region$data_dir
-    region$repopulate_region
 
-    to_plot$straight_line <<- readRDS(file.path(region$data_dir, "l.Rds"))
-    to_plot$zones <<- readRDS(file.path(region$data_dir, "z.Rds"))
-    to_plot$cents <<- readRDS(file.path(region$data_dir, "c.Rds"))
+    # Define region geography, forcing a default in cases where the geography is not available
+    if (input$purpose =="commute") {
+      if (input$geography %in% c("msoa", "lsoa")) {
+        region$geography <<- input$geography
+      } else {
+        region$geography <<- "msoa"
+      }
+    } else if (input$purpose =="school") {
+      if (input$geography %in% c("lsoa")) {
+        region$geography <<- input$geography
+      } else {
+        region$geography <<- "lsoa"
+      }
+    } else if (input$purpose =="alltrips") {
+      if (input$geography %in% c("msoa")) {
+        region$geography <<- input$geography
+      } else {
+        region$geography <<- "msoa"
+      }
+    }
 
-    to_plot$route_network <<- readRDS(file.path(region$data_dir, "rnet.Rds"))
-    to_plot$route_network$id <<- 1:nrow(to_plot$route_network)
+    # Set data_dir
+    region$data_dir <<- file.path(data_regional_root, input$purpose, region$geography, region$current)
 
-    to_plot$faster_route <<- readRDS(file.path(region$data_dir, "rf.Rds" ))
-    to_plot$faster_route@data <<- cbind(
-      to_plot$faster_route@data[!(names(to_plot$faster_route) %in% names(to_plot$straight_line))],
-      to_plot$straight_line@data)
-    to_plot$quieter_route <<- readRDS(file.path(region$data_dir, "rq.Rds"))
-    to_plot$quieter_route@data <<- cbind(
-      to_plot$quieter_route@data[!(names(to_plot$quieter_route) %in% names(to_plot$straight_line))],
-      to_plot$straight_line@data)
+    # Identify that region repopulation has happened
+    region$repopulate_region <<- T
 
-    # Add rqincr column to the quiet data
-    to_plot$quieter_route@data$rqincr <<- to_plot$quieter_route@data$length / to_plot$faster_route@data$length
+    # Identify purposes and geographies available in region
+    purposes_list <- c("commute", "school", "alltrips")
+    region$purposes_present <<- (dir.exists(file.path(data_regional_root, purposes_list, "msoa", region$current)) | dir.exists(file.path(data_regional_root, purposes_list, "lsoa", region$current)))
+    geographies_list <- c("msoa", "lsoa")
+    region$geographies_present <<- dir.exists(file.path(data_regional_root, input$purpose, geographies_list, region$current))
```

Comment:
So `region$geographies_present` is now a column vector?

Reply:
Also I don't think this needs to be stored in the `region` globally as it isn't used elsewhere.

Reply:
oh I see, you mean one could create it in a temporary way and then feed it ddirectlyinto 'update_purposegeog' function without saving it? yes sure, if that works then great

Reply:
Given the ∞ loop issue I'll leave this as it is!